src/main/java/com/worksap/nlp/sudachi/dictionary/WordInfo.java

```diff
@@ -253,13 +253,13 @@ private WordInfo(ByteBuffer buffer, int pos) {
         // do not modify buffer metadata for better performance
         posId = buffer.getShort(pos + 6);
 
-        surface = surfaceForm(buffer, pos); // +8
+        headword = headwordForm(buffer, pos); // +8
         reading = readingForm(buffer, pos); // +12
         normalizedForm = buffer.getInt(pos + 16);
         dictionaryForm = buffer.getInt(pos + 20);
 
         long rest = buffer.getLong(pos + 24);
-        headwordLength = (short) (rest & 0xffff);
+        indexFormLength = (short) (rest & 0xffff);
```

Comment:
Which length should be in the dictionary, indexform or headword?

Reply:
This must be a length of indexform since it is used during analysis, which is performed on the bytes of normalized input.
I will improve comments to clear this point.